### ISSUE TITLE
Missing typo correction diagnostic in preprocessor conditionals

### ISSUE THREAD

**Issue by AaronBallman** [llvm/llvm-project#51598]
|  |  |
| --- | --- |
| Bugzilla Link | [52256](https://llvm.org/bz52256) |
| Version | trunk |
| OS | Windows NT |
| CC | @zygoloid |

## Extended Description 
Consider the following code, and assume that FOO is not defined:
```
#ifdef FOO
#elfidef BAR
#endif
```
In this case, we do not issue any diagnostic about the typo where `#elifdef` is misspelled as `#elfidef` because it appears within a preprocessor branch that is discarded.

When scanning for the end of the discarded block, we should consider typo corrections for non-directives to help catch this sort of mistake.
